src/controllers/IllnessStartDateController.ts

```diff
@@ -0,0 +1,75 @@
+import { FormValidator } from 'app/controllers/validators/FormValidator';
+import { IllnessReasonFeatureMiddleware } from 'app/middleware/IllnessReasonFeatureMiddleware';
+import { schema } from 'app/models/fields/IllnessStartDate.schema';
+import { ILLNESS_START_DATE_PAGE_URI } from 'app/utils/Paths';
+import { ValidationResult } from 'app/utils/validation/ValidationResult';
+
+const template = 'illness/illness-start-date';
+const ILLNESS_START_DAY_FIELD = 'startDay';
+const ILLNESS_START_MONTH_FIELD = 'startMonth';
+const ILLNESS_START_YEAR_FIELD = 'startYear';
+const ILLNESS_FULL_DATE = 'startDate';
+
+@controller(ILLNESS_START_DATE_PAGE_URI, SessionMiddleware, IllnessReasonFeatureMiddleware)
+export class IllnessStartDateController extends BaseAsyncHttpController {
+
+    @httpGet('')
+    public async renderView(): Promise<void> {
+        return this.render(template);
+    }
+
+    @httpPost('')
+    public async continue(): Promise<void> {
+
+        const request = this.httpContext.request;
+
+        request.body.startDate = new Date(
+            `${request.body[ILLNESS_START_YEAR_FIELD]}-${request.body[ILLNESS_START_MONTH_FIELD]}-${request.body[ILLNESS_START_DAY_FIELD]}`);
+
+        let startDateDayErrorFlag: boolean = false;
+        let startDateMonthErrorFlag: boolean = false;
+        let startDateYearErrorFlag: boolean = false;
+
+        const validationResult: ValidationResult = await new FormValidator(schema).validate(request);
+
+        if (validationResult.errors.length > 0) {
+
+            validationResult.errors.forEach(err => {
+
+                switch (err.field) {
+                    case ILLNESS_START_DAY_FIELD:
+                        startDateDayErrorFlag = true;
+                        break;
+                    case ILLNESS_START_MONTH_FIELD:
+                        startDateMonthErrorFlag = true;
+                        break;
+                    case ILLNESS_START_YEAR_FIELD:
+                        startDateYearErrorFlag = true;
+                        break;
+                    case ILLNESS_FULL_DATE:
+                        if (startDateDayErrorFlag || startDateMonthErrorFlag || startDateYearErrorFlag) {
+                            validationResult.errors.splice(validationResult.errors.indexOf(err), 1);
+                        }
+                        startDateDayErrorFlag = true;
+                        startDateMonthErrorFlag = true;
+                        startDateYearErrorFlag = true;
+                        break;
+                }
+            });
+            return this.renderWithStatus(UNPROCESSABLE_ENTITY)(
+                template, {
+                    validationResult,
+                    isStartDateDayErrorFlag: startDateDayErrorFlag,
+                    isStartDateMonthErrorFlag: startDateMonthErrorFlag,
+                    isStartDateYearErrorFlag: startDateYearErrorFlag,
+                });
```

Comment:
In your screenshots, the errors seem to be displayed correctly, but the invalid data that the user previously would have put in, is erased. This behavior is different to how we implemented the penalty reference page, example bellow:

![Screenshot 2020-07-27 at 13 28 46](https://user-images.githubusercontent.com/17221386/88541787-3f195b00-d00d-11ea-98cb-7d0432fe99f6.png)

Reply:
this should have been resolved now